x/genutil/gentx.go

```diff
@@ -100,17 +100,17 @@ func DeliverGenTxs(
 	for _, genTx := range genTxs {
 		tx, err := txEncodingConfig.TxJSONDecoder()(genTx)
 		if err != nil {
-			panic(err)
+			panic(fmt.Sprintf("failed to decode gentx: %s error: %s", genTx, err))
```

Comment:
can we please return error instead of panicing?

Reply:
We can do that but `DeliverGenTxs` is called inside the `InitGenesis`, it will panic there on error anyways. I don't see a benefit for this case

Reply:
We should panic only when necessary.
This function already returns error. So the `InitGenesis` can decide if it will panic or not - it must handle the error anyway